apollo-router/src/configuration/mod.rs

```diff
@@ -1109,6 +1109,19 @@ impl From<SocketAddr> for ListenAddr {
     }
 }
 
+#[allow(clippy::from_over_into)]
+impl Into<serde_json::Value> for ListenAddr {
+    fn into(self) -> serde_json::Value {
+        match self {
+            // It avoids to prefix with `http://` when serializing and relying on the Display impl.
+            // Otherwise, it's converted to a `UnixSocket` in any case.
+            Self::SocketAddr(addr) => serde_json::Value::String(addr.to_string()),
+            #[cfg(unix)]
+            Self::UnixSocket(path) => serde_json::Value::String(format!("{}", path.display())),
```

Comment:
If the path is not UTF-8 `.display()` will produce something that won’t match when passing it back to a system call. Instead use `path.as_os_str().to_str().expect("unsupported non-UTF-8 path")` to panic in that case (or propagate the error with a `TryInto` impl instead of `Into`)

Reply:
in rover we use the [`camino`](https://crates.io/crates/camino) crate to handle file paths - they provide a `Utf8PathBuf` and `Utf8Path` types that require a path to be valid UTF-8 to create along with handy errors that you can propagate up. not a hard requirement here but thought you may find it interesting

Reply:
I’d be on board with using camino, but if so we should switch the entire code base in a dedicated PR.